consensus/vbft/service.go

```diff
@@ -561,16 +576,22 @@ func (self *Server) run(peerPubKey keypair.PublicKey) error {
 		// TODO: handle peer disconnection here
 
 		log.Warnf("server %d: disconnected with peer %d", self.Index, peerIdx)
-		close(self.msgRecvC[peerIdx])
-		delete(self.msgRecvC, peerIdx)
-
-		self.peerPool.peerDisconnected(peerIdx)
-		self.stateMgr.StateEventC <- &StateEvent{
-			Type: UpdatePeerState,
-			peerState: &PeerState{
-				peerIdx:   peerIdx,
-				connected: false,
-			},
+		if _, present := self.msgRecvC[peerIdx]; present {
+			log.Infof("close chan msgRecv peer:%d", peerIdx)
+			close(self.msgRecvC[peerIdx])
+			delete(self.msgRecvC, peerIdx)
+		}
+		if self.peerPool.getPeer(peerIdx) != nil {
+			self.peerPool.peerDisconnected(peerIdx)
```

Comment:
? If peer not in peer pool, when started the peer?

Reply:
No.but when we stop consensus,we can clean peerpool. at the same time server disconnected with other peer, invoke defer fun() method.